std/net.zig

```diff
@@ -15,19 +15,28 @@ pub const OsAddress = switch (builtin.os) {
     else => posix.sockaddr,
 };
 
+pub const OsAddressLength = switch (builtin.os) {
+    builtin.Os.windows => u32,
+    else => std.os.posix.socklen_t,
+};
+
 pub const Address = struct {
     os_addr: OsAddress,
+    os_length: OsAddressLength,
 
+    /// Assumes that ip4 address is in host byte order
     pub fn initIp4(ip4: u32, port: u16) Address {
         return Address{
             .os_addr = posix.sockaddr{
-                .in = posix.sockaddr_in{
+                .in = posix.sockaddr_in {
+                    .len = @sizeOf(posix.sockaddr_in),
                     .family = posix.AF_INET,
                     .port = std.mem.endianSwapIfLe(u16, port),
-                    .addr = ip4,
+                    .addr = std.mem.endianSwapIfLe(u32, ip4),
```

Comment:
net.parseIp4() assumes ip addresses are big-endian.

Reply:
@shawnl Thanks. I redid net.parseIp4 in the update. I am cleaning it up before I post the next WIP.